Breaking change with requiring `npm 7` without bumping the major version

This PR https://github.com/apollographql/apollo-client/commit/68d9cabab976a4ca659e161ec52ea1700aab8496 is causing issues with our repo. 
```
➜  unified-checkout-v3 git:(gpinto/ECOMM-106839_sessionTimeOut_emitMessage) ✗ npm i
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for @apollo/client@3.4.3: wanted: {"npm":"^7.20.3"} (current: {"node":"12.14.1","npm":"6.14.13"})
npm ERR! notsup Not compatible with your version of node/npm: @apollo/client@3.4.3
npm ERR! notsup Not compatible with your version of node/npm: @apollo/client@3.4.3
npm ERR! notsup Required: {"npm":"^7.20.3"}
npm ERR! notsup Actual:   {"npm":"6.14.13","node":"12.14.1"}
```
Why enforcing the npm version?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+## Apollo Client 3.4.4
+
+### Bug Fixes
+
+- Revert accidental addition of `engines.npm` section to published version of `@apollo/client/package.json`. <br/>
+  [@benjamn](https://github.com/benjamn) in [#8578](https://github.com/apollographql/apollo-client/pull/8578)
+
 ## Apollo Client 3.4.3
 
 ### Bug Fixes

--- a/config/prepareDist.js
+++ b/config/prepareDist.js
@@ -31,6 +31,7 @@ packageJson.private = false;
 // Remove package.json items that we don't need to publish
 delete packageJson.scripts;
 delete packageJson.bundlesize;
+delete packageJson.engines;
 
 // The root package.json points to the CJS/ESM source in "dist", to support
 // on-going package development (e.g. running tests, supporting npm link, etc.).

